basex.go

```diff
@@ -49,29 +49,16 @@ func Encode(s string) (string, error) {
 	var strVal string
 
 	a := big.NewInt(0)
-	b := big.NewInt(0)
-	c := big.NewInt(0)
-	d := big.NewInt(0)
-
-	exponent := 1
-
 	remaining := big.NewInt(0)
 	remaining.SetString(s, 10)
 
 	for remaining.Cmp(big.NewInt(0)) != 0 {
-		a.Exp(base, big.NewInt(int64(exponent)), nil) //16^1 = 16
-		b = b.Mod(remaining, a)                       //119 % 16 = 7 | 112 % 256 = 112
-		c = c.Exp(base, big.NewInt(int64(exponent-1)), nil)
-		d = d.Div(b, c)
-
-		//if d > dictionary.length, we have a problem. but BigInteger doesnt have
-		//a greater than method :-(  hope for the best. theoretically, d is always
-		//an index of the dictionary!
-		strVal = d.String()
+		a.Mod(remaining, base)
```

Comment:
I understand this is working. How do you say that this is better than the old code? Could you explain this approach?

Reply:
Looks to me like the math is just being simplified down into fewer steps, and a lot less code.